(GH-2013) Update docs on PowerShell execution policy (#2501)
puppetlabs/bolt: documentation/bolt_installing.md

### Change execution policy restrictions

PowerShell's execution policy is a safety feature that controls the conditions
under which PowerShell loads configuration files and runs scripts. This feature
helps prevent the execution of malicious scripts. The default policy is
`Restricted` (allow no scripts to run) for Windows _clients_ and `RemoteSigned`
(allow signed scripts and non-signed scripts not from the internet) for Windows
_servers_. Some environments change this to `AllSigned`, which only allows
scripts to run as long as they are signed from with a trusted publisher.

> As of Bolt 2.21.0, we sign the Bolt PowerShell Module files with a Puppet code
> signing cert. This means that environments that have `AllSigned` set will allow
> the `bolt` command to work without user input so long as they trust our
> publisher. If they do not implicitly trust the publisher, they can accept the
> publisher without having to change `ExecutionPolicy`. Environments that have it
> set to `Restricted` will still need to change their policy to `RemoteSigned` or
> `AllSigned`. These restrictions are easy to change, but check with your security
> team first.

Reviewer: ```suggestion
As of Bolt 2.21.0, we sign Bolt PowerShell module files with a Puppet code
signing certificate. If your PowerShell environment uses an `AllSigned` execution 
policy and you add Puppet as a trusted publisher, the `bolt` command works 
without any further input. If you're using the `AllSigned` policy and you have 
not added Puppet as a trusted publisher, you can accept the publisher without 
having to change your execution policy.

If your environment uses a `Restricted` policy, you must change your policy to 
`RemoteSigned` or`AllSigned`. Check with your security team before you make 
any policy changes.
```

Reviewer: Question: How would one add Puppet as a trusted publisher? Talk to your security team?

Author: My understanding is that you're considered a trusted publisher if the script is signed with a valid, unexpired digital signature issued by a certificate authority. @jpogran would know better since he's much more familiar with Windows, but I don't think we need to cover that since it only applies to users running old versions of Bolt.

Reviewer: @beechtom After consulting with James, I don't think we need to add anything else to this edit 👍

> If you see this or a similar error when trying to run Bolt, you probably need to
> change your script execution policy restrictions, as described here.

Reviewer: ```suggestion
If you see this or a similar error when trying to run Bolt, you probably need to
change your script execution policy restrictions:
```

                n
                + FullyQualifiedErrorId : CouldNotAutoloadMatchingModule
```

> 

Reviewer: ```suggestion
To change your script execution policy:
```
